hibernate-core/src/main/java/org/hibernate/tool/schema/extract/internal/InformationExtractorJdbcDatabaseMetaDataImpl.java

```diff
@@ -282,7 +282,7 @@ private TableInformation locateTableInNamespace(
 
 		if ( extractionContext.getJdbcEnvironment().getNameQualifierSupport().supportsCatalogs() ) {
 			if ( catalog == null ) {
-				catalogFilter = "";
+				catalogFilter = null;
```

Comment:
According to [DatabaseMetaData.html#getColumns](https://docs.oracle.com/javase/8/docs/api/java/sql/DatabaseMetaData.html#getColumns-java.lang.String-java.lang.String-java.lang.String-java.lang.String-) documentation, the `catalog` argument is interpreted as follows:

> catalog - a catalog name; must match the catalog name as it is stored in the database; "" retrieves those without a catalog; null means that the catalog name should not be used to narrow the search

So, if the database supports catalogs, using `""` will filter those that do not have any catalog, while `null` means that the catalog is not taken into consideration.

Which database do you consider this default setting is not working properly with?